api/src/main/java/org/apache/iceberg/util/StructProjection.java

```diff
@@ -174,6 +174,7 @@ private StructProjection(StructType structType, StructType projection, boolean a
   }
 
   public StructProjection wrap(StructLike newStruct) {
+    Preconditions.checkState(newStruct != null, "Invalid root struct for wrapping: null");
```

Comment:
I feel the removal of if-check and adding Preconditions check should go hand-to-hand. Either we do both or don't change it all. Otherwise, we can get NPE for wrapping root null object.

One Preconditions check shouldn't cause performance degradation. Overall, we removed one if check and added a new check in Preconditions.

Here are a few usages on the `StructProjection#wrap(StructLike)` 
* core: transform on `CloseableIterable<StructLike>` for manifest or file entries
* data: delete filter on record
* flink: always wrap non-null `RowDataWrapper` object for EqualityFieldKeySelector
* spark: always wrap non-null `InternalRowWrapper` object for wrapping partition data

@RussellSpitzer @rdblue do you see any potential null root values for the core and data module usage of wrap.

Reply:
The `struct` field is defaulted to `null`. Plus, a check here would create a failure rather than gracefully handling a null struct. If the struct is somehow null, then the projection should be as well right?

Reply:
@rdblue you are saying wrapping a null root struct should be allowed (not throwing exception), right?

```
StructProjection projection = StructProjection.create(schema, projectedSchema);
projection.wrap(null); // allowed
```